mmstudio/src/main/java/org/micromanager/acquisition/internal/acqengjcompat/AcqEngJAdapter.java

```diff
@@ -482,14 +555,94 @@ private Iterator<AcquisitionEvent> createAcqEventIterator(SequenceSettings acqui
+                     && event.getAxisPosition("position") != null) {
+               if (startTime_ == 0) {
+                  startTime_ = System.currentTimeMillis();
+               }
+
+               int thisPosition = (int) event.getAxisPosition("position");
+               if (thisPosition != lastPositionIndex_) {
+                  relativePositionStartTime_ =  System.currentTimeMillis() - startTime_;
+                  System.out.println("Position " + thisPosition + " started "
+                          + relativePositionStartTime_ + "  ms after acquisition start");
+                  lastPositionIndex_ = (int) event.getAxisPosition("position");
+                  positionMoved_ = true;
+               }
+               if (positionMoved_) {
+                  long relativeStartTime = relativePositionStartTime_
+                          + event.getMinimumStartTimeAbsolute() - startTime_;
+                  int frame = (int) event.getAxisPosition("time");
+                  System.out.println("Pos " + thisPosition + ", Frame " + frame
+                          + " start at " + relativeStartTime);
+                  event.setMinimumStartTime(relativeStartTime);
+               }
+            }
+             */
+            if (event.getMinimumStartTimeAbsolute() != null) {
+               nextWakeTime_ = event.getMinimumStartTimeAbsolute();
+            }
+            return event;
+         }
+      };
+   }
+
+   private AcquisitionHook timeLapseHook(SequenceSettings sequenceSettings) {
+      return new AcquisitionHook() {
+         private int lastPositionIndex_ = 0;
+         private long relativePositionStartTime_ = 0;
+         private long startTime_ = 0;
+         private boolean positionMoved_ = false;
+
+         @Override
+         public AcquisitionEvent run(AcquisitionEvent event) {
+            if (sequenceSettings.acqOrderMode() == AcqOrderMode.POS_TIME_CHANNEL_SLICE
+                  || sequenceSettings.acqOrderMode() == AcqOrderMode.POS_TIME_SLICE_CHANNEL
+                  && event.getAxisPosition("position") != null) {
+               if (startTime_ == 0) {
+                  startTime_ = System.currentTimeMillis();
+               }
+
+               int thisPosition = (int) event.getAxisPosition("position");
+               if (thisPosition != lastPositionIndex_) {
+                  relativePositionStartTime_ =  System.currentTimeMillis() - startTime_;
+                  System.out.println("Position " + thisPosition + " started "
+                        + relativePositionStartTime_ + "  ms after acquisition start");
+                  lastPositionIndex_ = (int) event.getAxisPosition("position");
+                  positionMoved_ = true;
+               }
+               if (positionMoved_) {
+                  long relativeStartTime = relativePositionStartTime_
+                        + event.getMinimumStartTimeAbsolute() - startTime_;
```

Comment:
Is `event.getMinimumStartTimeAbsolute()` actually giving the absolute start time? Or is it relative?